hs_core/hydroshare/resource.py

```diff
@@ -369,31 +369,30 @@ def create_resource(
 
         add_resource_files(resource.short_id, *files)
 
-        if 'owner' in kwargs:
-            owner = utils.user_from_id(kwargs['owner'])
-            resource.owners.add(owner)
-
-        resource.owners.add(owner)
+        resource_access = ResourceAccess(resource=resource)
+        resource_access.save()
+        UserResourcePrivilege(resource=resource_access, grantor=owner.uaccess, user=owner.uaccess,
+                              privilege=PrivilegeCodes.OWNER).save()
 
         if edit_users:
             for user in edit_users:
                 user = utils.user_from_id(user)
-                resource.edit_users.add(user)
```

Comment:
In general, I suggest running old and new access control side by side until views are integrated. Feel free to mark old access control code for deletion/deprecation. But I think that integrating the views is going to take a while. If we have the time, then this is fine. I do not know the schedule requirements.